cmd/gateway/s3x/gateway-s3x-multipart.go

```diff
@@ -46,15 +51,18 @@ func (x *xObjects) PutObjectPart(
 	if err != nil {
 		return pi, x.toMinioErr(err, bucket, object, uploadID)
 	}
-	err = x.ledgerStore.PutObjectPart(bucket, object, hash, uploadID, int64(partID))
-	if err != nil {
-		return pi, x.toMinioErr(err, bucket, object, uploadID)
-	}
-	return minio.PartInfo{
+	pi = minio.PartInfo{
 		PartNumber:   partID,
 		LastModified: time.Now().UTC(),
+		ETag:         hash,
 		Size:         int64(size),
-	}, nil
+		ActualSize:   int64(size),
+	}
+	err = x.ledgerStore.PutObjectPart(bucket, object, uploadID, pi)
```

Comment:
This should be:

```Go
if err = x.ledgerStore.PutObjectPart(bucket, object, uploadID, pi); err != nil {
    return pi, x.toMinioErr(err, bucket, object, uploadID)
```
In general I like this pattern for when we are simply checking the returned error.

Alternative we an condense into:

```GO
return pi, x.toMinioErr(
  x.ledgerStore.PutObjectPart(bucket, object, uploadID, pi),
bucket, object, uploadID,
)
```

Reply:
ok